bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -462,6 +463,11 @@ async fn extension_loop_active(
                                             stats_flusher.manual_flush()
                                         );
                                     }
+
+                                    if let Some(offsets) = enhanced_metric_data {
```

Comment:
This seems to be done after the flushing, a few lines above. It seems that potentially we will miss flushing these metrics.
Is it intended?